.github/test-build

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+
+# Remember where we started; this is where our source checkout lives.
+src=`pwd`
+
+# Load the module command and use it to load MPI.
+. /etc/profile.d/modules.sh
+module load mpi
+
+# Configure sccache to use a docker volume
+export SCCACHE_DIR=/sccache
+export SCCACHE_CACHE_SIZE="1G"
+echo "::group::Setup"
+echo "Cache size: "
+du -sh /sccache
+sccache -s
+echo "::endgroup::"
+
+# Create a directory to hold the build, then build and test IBAMR
+mkdir /bld
+cd /bld
+echo "::group::Configure"
+cmake \
+  -DCMAKE_C_COMPILER_LAUNCHER:STRING=sccache \
+  -DCMAKE_CXX_COMPILER_LAUNCHER:STRING=sccache \
+  -DCMAKE_Fortran_COMPILER_LAUNCHER:STRING=sccache \
+  -DCMAKE_C_FLAGS="-O3 -fPIC -march=native" \
+  -DCMAKE_CXX_FLAGS="-O3 -fPIC -march=native" \
+  -DCMAKE_Fortran_FLAGS="-O3 -fPIC -march=native" \
+  -DCMAKE_INSTALL_PREFIX=/ibamr \
+  -DSAMRAI_ROOT=/samrai \
+  -DLIBMESH_ROOT=/libmesh \
+  -DLIBMESH_METHOD=OPT \
+  -DPETSC_ROOT=/petsc/x86_64 \
+  -DHYPRE_ROOT=/petsc/x86_64 \
+  -DNUMDIFF_ROOT=/numdiff \
+  ${src}
+echo "::endgroup::"
+echo "::group::Build"
+time make -j4
+sccache -s
+echo "::endgroup::"
+echo "::group::Test install"
+time make -j4 install
+echo "::endgroup::"
+echo "::group::Build tests"
+time (cd tests; make -j4 tests)
+sccache -s
+echo "::endgroup::"
+echo "::group::Run tests"
+echo "Tests disabled until hanging tests are fixed."
```

Comment:
I'll add a timeout to `attest` so we can get this in.

Reply:
Fixed by #1287.

Reply:
I've rebased to catch the fix, enabled the tests, and pushed; we'll see what happens with the tests.

Reply:
We can remove that last `echo` now that tests are back on.

Reply:
Oops, I'll fix it after the action runs to completion on the [other PR](https://github.com/vibraphone/IBAMR/runs/3049643028?check_suite_focus=true)...

Reply:
Oh, cool, it's actually running somewhere!

Reply:
Yes, if you look at the actions tab in my fork, you can see the long string of failures it took to get here. 😄